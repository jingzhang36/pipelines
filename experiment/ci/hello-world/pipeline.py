#!/usr/bin/env python3
# Copyright 2019 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.


import kfp.dsl as dsl
import argparse

parser = argparse.ArgumentParser()
parser.add_argument('--commit_id', help='Commit Id', type=str)
args = parser.parse_args()

@dsl.pipeline(
    name='Hello World CI pipeline',
    description='Basic sample to show how to build docker image as part of KFP CI'
)
def helloworld_ci_pipeline():
  dsl.ContainerOp(
      name='Print Hello World',
      image='gcr.io/ml-pipeline-dogfood/helloworld-ci:' + args.commit_id,
  )


if __name__ == '__main__':
  import kfp.compiler as compiler
  compiler.Compiler().compile(helloworld_ci_pipeline, __file__ + '.zip')