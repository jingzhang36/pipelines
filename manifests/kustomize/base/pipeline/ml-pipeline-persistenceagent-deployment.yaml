apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: ml-pipeline-persistenceagent
  name: ml-pipeline-persistenceagent
spec:
  selector:
    matchLabels:
      app: ml-pipeline-persistenceagent
  template:
    metadata:
      labels:
        app: ml-pipeline-persistenceagent
    spec:
      containers:
      - env:
          - name: NAMESPACE
            valueFrom:
              fieldRef:
                fieldPath: metadata.namespace
          # If you want a different TTL_SECONDS_AFTER_WORKFLOW_FINISH than the
          # default 1 day, please specify it here as demonstrated below.
          # - name: TTL_SECONDS_AFTER_WORKFLOW_FINISH
          #   value: "172800"
        image: gcr.io/ml-pipeline/persistenceagent:dummy
        imagePullPolicy: IfNotPresent
        name: ml-pipeline-persistenceagent
      serviceAccountName: ml-pipeline-persistenceagent
