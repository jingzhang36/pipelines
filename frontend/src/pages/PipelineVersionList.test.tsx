/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from 'react';
import PipelineVersionList, { PipelineVersionListProps } from './PipelineVersionList';
import TestUtils from '../TestUtils';
import { ApiPipeline, ApiPipelineVersion } from '../apis/pipeline';
import { Apis, ListRequest } from '../lib/Apis';
import { PageProps } from './Page';
import { RoutePage, RouteParams } from '../components/Router';
import { shallow, ReactWrapper, ShallowWrapper } from 'enzyme';
import { range } from 'lodash';
import { ImportMethod } from '../components/UploadPipelineDialog';
import { ButtonKeys } from '../lib/Buttons';

class PipelineVersionListTest extends PipelineVersionList {
  public _loadPipelineVersions(request: ListRequest): Promise<string> {
    return super._loadPipelineVersions(request);
  }
}

describe('PipelineVersionList', () => {
  let tree: ReactWrapper | ShallowWrapper;

  const listPipelineVersionsSpy = jest.spyOn(Apis.pipelineServiceApi, 'listPipelineVersions');
  // const createPipelineVersionSpy = jest.spyOn(Apis.pipelineServiceApi, 'createPipelineVersion');
  // const deletePipelineVersionSpy = jest.spyOn(Apis.pipelineServiceApi, 'deletePipelineVersion');
  const onErrorSpy = jest.fn();


  function generateProps(): PipelineVersionListProps {
    return {
      history: {} as any,
      location: { search: '' } as any,
      match: '' as any,
      onError: onErrorSpy,
    };
  }

  async function mountWithNPipelineVersions(n: number): Promise<ReactWrapper> {
    listPipelineVersionsSpy.mockImplementationOnce((pipelineId: string) => ({
      pipelineVersions: range(n).map(i => ({
        id: 'test-pipeline-version-id' + i,
        name: 'test pipeline version name' + i,
      })),
    }));
    tree = TestUtils.mountWithRouter(<PipelineVersionList {...generateProps()} />);
    await listPipelineVersionsSpy;
    await TestUtils.flushPromises();
    tree.update(); // Make sure the tree is updated before returning it
    return tree;
  }

  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterEach(async () => {
    // unmount() should be called before resetAllMocks() in case any part of the unmount life cycle
    // depends on mocks/spies
    await tree.unmount();
    jest.resetAllMocks();
  });

  it('renders an empty list with empty state message', () => {
    tree = shallow(<PipelineVersionList {...generateProps()} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders a list of one pipeline version', async () => {
    tree = shallow(<PipelineVersionList {...generateProps()} />);
    tree.setState({
      pipelineVersions: [
        {
          created_at: new Date(2018, 8, 22, 11, 5, 48),
          name: 'pipelineversion1',
        } as ApiPipelineVersion,
      ],
    });
    await listPipelineVersionsSpy;
    expect(tree).toMatchSnapshot();
  });

  it('renders a list of one pipeline version without created date', async () => {
    tree = shallow(<PipelineVersionList {...generateProps()} />);
    tree.setState({
      pipelines: [
        {
          name: 'pipelineversion1',
        } as ApiPipelineVersion,
      ],
    });
    await listPipelineVersionsSpy;
    expect(tree).toMatchSnapshot();
  });

  it('renders a list of one pipeline version with error', async () => {
    tree = shallow(<PipelineVersionList {...generateProps()} />);
    tree.setState({
      pipelineVersions: [
        {
          created_at: new Date(2018, 8, 22, 11, 5, 48),
          error: 'oops! could not load pipeline',
          name: 'pipeline1',
          parameters: [],
        } as ApiPipelineVersion,
      ],
    });
    await listPipelineVersionsSpy;
    expect(tree).toMatchSnapshot();
  });

  it('calls Apis to list pipeline versions, sorted by creation time in descending order', async () => {
    // listPipelineVersionsSpy.mockImplementationOnce((pipelineId: string) => ({ pipelineVersions: [{ name: 'pipelineversion1' }] }));
    // tree = TestUtils.mountWithRouter(<PipelineVersionList {...generateProps()} />);
    // await listPipelineVersionsSpy;
    // expect(listPipelineVersionsSpy).toHaveBeenLastCalledWith('', 10, 'created_at desc', '');
    // expect(tree.state()).toHaveProperty('pipelineVersions', [{ name: 'pipelineversion1' }]);

    mountWithNPipelineVersions(1);
    const props = generateProps();
    tree = shallow(<PipelineVersionList {...props} />);
    await (tree.instance() as PipelineVersionListTest)._loadPipelineVersions({});
    expect(Apis.pipelineServiceApi.listPipelineVersions).toHaveBeenLastCalledWith('', 10, 'created_at desc', '');
  });
});
