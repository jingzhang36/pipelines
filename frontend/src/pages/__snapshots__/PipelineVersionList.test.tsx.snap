// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelineVersionList renders a list of one pipeline version 1`] = `
<div>
  <CustomTable
    columns={
      Array [
        Object {
          "customRenderer": [Function],
          "flex": 2,
          "label": "Version name",
          "sortKey": "name",
        },
        Object {
          "flex": 1,
          "label": "Uploaded on",
          "sortKey": "created_at",
        },
      ]
    }
    emptyMessage="No pipeline versions found."
    initialSortColumn="created_at"
    reload={[Function]}
    rows={
      Array [
        Object {
          "id": undefined,
          "otherFields": Array [
            "pipelineversion1",
            "9/22/2018, 11:05:48 AM",
          ],
        },
      ]
    }
  />
</div>
`;

exports[`PipelineVersionList renders a list of one pipeline version with error 1`] = `
<div>
  <CustomTable
    columns={
      Array [
        Object {
          "customRenderer": [Function],
          "flex": 2,
          "label": "Version name",
          "sortKey": "name",
        },
        Object {
          "flex": 1,
          "label": "Uploaded on",
          "sortKey": "created_at",
        },
      ]
    }
    emptyMessage="No pipeline versions found."
    initialSortColumn="created_at"
    reload={[Function]}
    rows={
      Array [
        Object {
          "id": undefined,
          "otherFields": Array [
            "pipeline1",
            "9/22/2018, 11:05:48 AM",
          ],
        },
      ]
    }
  />
</div>
`;

exports[`PipelineVersionList renders a list of one pipeline version without created date 1`] = `
<div>
  <CustomTable
    columns={
      Array [
        Object {
          "customRenderer": [Function],
          "flex": 2,
          "label": "Version name",
          "sortKey": "name",
        },
        Object {
          "flex": 1,
          "label": "Uploaded on",
          "sortKey": "created_at",
        },
      ]
    }
    emptyMessage="No pipeline versions found."
    initialSortColumn="created_at"
    reload={[Function]}
    rows={Array []}
  />
</div>
`;

exports[`PipelineVersionList renders a list of one pipeline versions without created date 1`] = `
<div>
  <CustomTable
    columns={
      Array [
        Object {
          "customRenderer": [Function],
          "flex": 2,
          "label": "Version name",
          "sortKey": "name",
        },
        Object {
          "flex": 1,
          "label": "Uploaded on",
          "sortKey": "created_at",
        },
      ]
    }
    emptyMessage="No pipeline versions found."
    initialSortColumn="created_at"
    reload={[Function]}
    rows={Array []}
  />
</div>
`;

exports[`PipelineVersionList renders a list of one pipeline with error 1`] = `
<div>
  <CustomTable
    columns={
      Array [
        Object {
          "customRenderer": [Function],
          "flex": 2,
          "label": "Version name",
          "sortKey": "name",
        },
        Object {
          "flex": 1,
          "label": "Uploaded on",
          "sortKey": "created_at",
        },
      ]
    }
    emptyMessage="No pipeline versions found."
    initialSortColumn="created_at"
    reload={[Function]}
    rows={Array []}
  />
</div>
`;

exports[`PipelineVersionList renders an empty list with empty state message 1`] = `
<div>
  <CustomTable
    columns={
      Array [
        Object {
          "customRenderer": [Function],
          "flex": 2,
          "label": "Version name",
          "sortKey": "name",
        },
        Object {
          "flex": 1,
          "label": "Uploaded on",
          "sortKey": "created_at",
        },
      ]
    }
    emptyMessage="No pipeline versions found."
    initialSortColumn="created_at"
    reload={[Function]}
    rows={Array []}
  />
</div>
`;
